type-c-service/src/service/config.rs

```diff
@@ -1,10 +1,90 @@
 use embedded_usb_pd::ucsi;
 
+/// UCSI battery charging capability status configuration
+///
+/// The implementation checks these thresholds in order from not_battery_charging_mw to slow_battery_charging_mw.
+/// Misordering the thresholds will lead to the first matching threshold being selected.
+#[derive(Debug, Clone, Copy, Default)]
+pub struct UcsiBatteryChargingThresholdConfig {
+    /// No battery charging power threshold in milliwatts
+    ///
+    /// Below this level GET_CONNECTOR_STATUS will report no battery charging
+    pub not_battery_charging_mw: Option<u32>,
+    /// Very slow battery charging power threshold in milliwatts
+    ///
+    /// Below this level GET_CONNECTOR_STATUS will report very slow battery charging
+    pub very_slow_battery_charging_mw: Option<u32>,
+    /// Slow battery charging power threshold in milliwatts
+    ///
+    /// Below this level GET_CONNECTOR_STATUS will report slow battery charging
+    pub slow_battery_charging_mw: Option<u32>,
+}
+
+impl UcsiBatteryChargingThresholdConfig {
+    /// Try to create a new Self with validation
```

Comment:
Since the fields are public, there's no way to enforce the validation that this method performs. Getters + making the fields private would ensure that if an instance of `UcsiBatteryChargingThresholdConfig` exists, its fields are well-formed.